lib/heuristics/concerns/scheduling_data_initialisation.rb

```diff
@@ -311,4 +309,22 @@ def compute_capacities(quantities, vehicle, available_units = [])
 
     capacities
   end
+
+  def exist_common_multiplexer?(lapses)
+    # If lapses have common multiplexer then any of these
+    # services will be assigned at the same time as service
+    # with smallest lapse (and normaly biggest number of visits).
+    # If this is not the case, we can not guarantee
+    # same_point_day constraint.
+    lapses = lapses.compact.uniq
+    min_lapse = lapses.min
+
+    multiplexers = lapses.collect{ |lapse|
+      (2..min_lapse).select{ |multiplexer|
+        (lapse % multiplexer).zero?
+      }
+    }
+
+    multiplexers.size < 2 || multiplexers.first.any?{ |value| multiplexers.all?{ |set| set.include?(value) } }
```

Comment:
It feels like `multiplexers.size < 2` case only happen if the lapses array has 1 uniq element. In that case, we can `return true` earlier (just after `lapses = lapses.compact.uniq`), right ?

Reply:
Actually IF the goal of this function is to check the existence a common divisor for all values in the `lapses` array then we don't need to do any of these, we already have such a function in the `Filters` class 

```ruby
def has_common_divisor?(lapses)
  uniq_lapses = lapses.compact.uniq
  return true if uniq_lapses.size < 2 

  gcd = Filters.gcd_of_array(uniq_lapses)
  return gcd > 1 # if gcd is 1 that means there is no common divisor
end
```

Since I don't know the exact usage, I can't be sure but it looks like to me this should do what you want. What do you think @fonsecadeline ?

Reply:
I did not know about this function, perfect for me !